src/libs/PersonalDetailsUtils.js

```diff
@@ -63,10 +63,22 @@ function getAccountIDsByLogins(logins) {
     return _.reduce(
         logins,
         (foundAccountIDs, login) => {
-            const currentDetail = _.find(personalDetails, (detail) => detail.login === login);
+            let isUserWithNoLogin = false;
+            const currentDetail = _.find(personalDetails, (detail) => {
+                if (!_.isUndefined(detail.login)) {
+                    isUserWithNoLogin = true;
+                }
+                return detail.login === login;
+            });
             if (!currentDetail) {
-                // generate an account ID because in this case the detail is probably new, so we don't have a real accountID yet
-                foundAccountIDs.push(UserUtils.generateAccountID(login));
+                // If the account login does not match login and the account login is undefined, we check if account display name matches login
+                const currentDetailWithNoLogin = _.find(personalDetails, (detail) => detail.displayName === login);
+                if (isUserWithNoLogin && currentDetailWithNoLogin) {
+                    foundAccountIDs.push(Number(currentDetailWithNoLogin.accountID));
+                } else {
+                    // generate an account ID because in this case the detail is probably new, so we don't have a real accountID yet
```

Comment:
```suggestion
                    // Generate an account ID because in this case the detail is probably new, so we don't have a real accountID yet
```